Run

letulip opened the pull request:
#Main метод Program класс
1. приложение принимает через аргументы два параметра mainCount и threadCount
* целое число для основного потока
* целое число для дополнительного потока
2. печатает входящие параметры
3. создаёт два потока
4. в цикле основного потока начинает выводить все числа от 1 до mainCount
5. в цикле дополнительного потока начинает выводить все числа от 1 до threadCount

#myThread класс
1. создает потоки
2. функция `func` потока обрабатывает входные данные с Mythread и производит вывод в консоль.

letulip committed:
AppConfig, Assemblyinfo добавлены

letulip committed:
Приложение принимает и печатает входные параметры

1. приложение принимает через аргументы два параметра mainCount и threadCount
* целое число для основного потока
* целое число для дополнительного потока
2. печатает входящие параметры

letulip committed:
myThread класс добавлен

В классе конструктор myThread получает имя функции и номер до кторого ведется счет.
Функция func потока обрабатывает входные данные с Mythread и производит вывод в консоль.

letulip committed:
В Main создаются потоки и выводятся числа

* создаёт дополнительный поток
* в цикле основного потока начинает выводить все числа от 0 до mainCount
* в цикле дополнительного потока начинает выводить все числа от 0 до threadCount

letulip committed:
Исправлен вывод чисел в func

Отсчет ведется с 1 вместо 0.

BekoBou commented:
Примеры вывода приложения можешь привести?

Входящие значения:

- 10, 10
- 3, 25

vldmkrsh commented on Program.cs:
```diff
@@ -0,0 +1,35 @@
+            Console.WriteLine("run.exe\nArguments:\n mainCount = {0}\n threadCount = {1}", mainCount, threadCount);
+
+            Console.WriteLine("Loops:");
+            myThread t1 = new myThread(" main", mainCount);
```
`main` поток — это именно тот поток, в котором ты создаёшь другой поток. Его не надо создавать.

letulip replied:
![5 10](https://cloud.githubusercontent.com/assets/23260217/23208977/42ba7904-f908-11e6-9886-2d9ac692dbde.png)
Пример вывода со значениями 5, 10
Потоки создаются в классе myThread, в методе Main я создаю два экземпляра класса, которы через функцию func в классе myThread осуществляют вывод значений

vldmkrsh commented on Program.cs:
```diff
@@ -0,0 +1,35 @@
+            myThread t1 = new myThread(" main", mainCount);
+            myThread t2 = new myThread(" thread", threadCount);
+
+            Console.ReadKey();
```
Не должно быть `Console.ReadKey();` в таких задачах. Запуск делай через терминал или окно откладки Visual Studio.

vldmkrsh replied:
У тебя здесь как раз должен быть цикл основного (`main`) потока.

letulip replied:
ReadKey() Находится в проекте для моего удобства и будет удален позже, мне нужно видеть что выводит программа.

letulip committed:
func считает до входного числа включительно

letulip committed:
Переделан Main в соответствии с замечаиями

vldmkrsh commented on Program.cs:
```diff
@@ -0,0 +1,43 @@
+            for (int i = 1; i <= (int)mainCount; i++)
+            {
+                Console.WriteLine(" main: " + i);
+                Thread.Sleep(0);
```
У тебя здесь «нет» `Thread`, надо убрать эту строку.

letulip replied:
Все отлично работает также, как и раньше =)

vldmkrsh commented:
Убери отмеченную строку, и попробуй запустить снова с разными параметрами.

letulip committed:
Удален Tread.sleep из Main

letulip committed:
Gitignore файлы

vldmkrsh commented on obj/Debug/.gitignore:
```diff
@@ -0,0 +1,4 @@
+/FixedThreadPool.csproj.FileListAbsolute.txt
+/TemporaryGeneratedFile_036C0B5B-1481-4323-8D20-8F5ADCB23D92.cs
```
ты можешь в корневой директории просто создать файл `.gitignore` с правилами вроде таких:

```
obj/Debug # игнорировать всю директорию Debug
TemporaryGeneratedFile*.cs # игнорировать все файлы, которые начинаются на TemporaryGeneratedFile и заканчиваются на .cs
```

Полный список правил и их написание ты всегда можешь найти в [документации](https://git-scm.com/docs/gitignore) или спросить.

letulip committed:
Добавлен Thread.Join() в конструктор myThread

Ветка, созданная через myThread встает в очередь и программа ждет ее выполнения.
